src/server/game/Spells/Spell.cpp

```diff
@@ -5483,10 +5483,17 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
             case SPELL_EFFECT_TALENT_SPEC_SELECT:
             {
-                ChrSpecializationEntry const* spec = sChrSpecializationStore.LookupEntry(m_misc.SpecializationId);
-                if (!spec || spec->ClassID != m_caster->getClass())
+                ChrSpecializationEntry const* spec = sChrSpecializationStore.LookupEntry(m_misc.SpecTree.SpecializationId);
+                Player* player = m_caster->ToPlayer();
+                Pet* pet = player->GetPet();
+
+                if (!spec || (spec->ClassID != m_caster->getClass() && spec->ClassID != 0))
                     return SPELL_FAILED_NO_SPEC;
 
+                if ((!spec || (spec->ClassID != m_caster->getClass() && spec->ClassID != 0 && m_misc.SpecTree.ClassId != 0)) &&
+                    (!pet || !pet->IsPet() || ((Pet*)player)->getPetType() != HUNTER_PET || pet->GetOwnerGUID() != player->GetGUID() || !pet->GetCharmInfo()))
```

Comment:
Tests performed:
>Build & Tested in-game

Your PR description doesn't mention it's  a WIP, it actually says it has been tested. If you are a new developer learning C++ please state it in the description so we know what we are expecting.

Everyone can still comment the PR even if closed so that makes very little difference to the feedback you can get.